### PR TITLE
Add `caughtErrorsIgnorePattern` option to `catch-error-name` rule

### LINKED ISSUE #48: Allow multiple `name` for `catch-error-name`
Sometimes we have the following code:

``` js
const fs = require('fs');
try {
  fs.readFileSync('./foo';)
} catch (err) {
  // Nothing
}
```

And if enable rule `no-unused-vars` we have error. For ignoring this construction we can use http://eslint.org/docs/rules/no-unused-vars#caughterrorsignorepattern. But if i use `ignoreErr` i get error from `catch-error-name` rule.

### PR DESCRIPTION
**Pull request by evilebottnawi**
Fixes #48
I think better add option which error name we should ignore (as do eslint) instead allow multiple errors names

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@sindresorhus friendly ping

File: docs/rules/catch-error-name.md [hunk 1]
```diff
@@ -75,8 +75,40 @@ somePromise.catch(_ => {
...
 "unicorn/catch-error-name": ["error", {"name": "err"}]
 ```
+
+### caughtErrorsIgnorePattern
+
+```js
+"unicorn/catch-error-name": ["error", {"caughtErrorsIgnorePattern": "^ignore"}]
+```
+
+The `caughtErrorsIgnorePattern` option specifies exceptions not to check for usage: catch arguments whose names match a regexp pattern.
```
Comment (sindresorhus) [inline docs/rules/catch-error-name.md]: This sentence could use some more work. It doesn't read very well.

File: rules/catch-error-name.js [hunk 2]
```diff
@@ -38,6 +38,9 @@ function indexifyName(name, scope) {
 const create = context => {
 	const opts = context.options[0];
 	const name = (opts && opts.name) || 'err';
```
Comment (sindresorhus) [inline rules/catch-error-name.js]: Use Object.assign so you don't have to check whether options exist or not each time:

```js
const opts = Object.assign({}, {name: 'err'}, context,options[0]);
```

**Comment by sindresorhus** [comment]
I think the `caughtErrorsIgnorePattern` option should default to `^_$`, so `catch (_) {}` is ignored by default (This should also be documented).

**Comment by sindresorhus** [comment]
Thanks for your patience. I've just been busy in real life.

**Comment by evilebottnawi** [comment]
@sindresorhus nothing terrible :star: , in the near future I will correct all comments

**Comment by evilebottnawi** [comment]
@sindresorhus default value was got from https://eslint.org/docs/rules/no-unused-vars#caughterrorsignorepattern, we can change, but i think better is use as in eslint or we can add multiple default value `_*` and `ignore*`

File: docs/rules/catch-error-name.md (hunk 1 above)
Comment (evilebottnawi) [reply docs/rules/catch-error-name.md]: @sindresorhus also was got from https://eslint.org/docs/rules/no-unused-vars#caughterrorsignorepattern. It is difficult to imagine how to explain in more detail

**Comment by sindresorhus** [comment]
> but i think better is use as in eslint or we can add multiple default value _* and ignore*

ESLint is generally very unopinionated, and I prefer good defaults. I meant it should be default, but the user could, of course, specify the option to get something else.

**Comment by sindresorhus** [comment]
Looks like you had a bad rebase.

**Comment by evilebottnawi** [comment]
@sindresorhus all done :+1:

**Comment by sindresorhus** [comment]
The default should be `^_$`, not `^_`.

**Comment by evilebottnawi** [comment]
@sindresorhus done :+1:

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**